include/debug/stack_size_analyzer.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#ifndef __STACK_SIZE_ANALYZER_H
+#define __STACK_SIZE_ANALYZER_H
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup stack_size_analyzer Stack analyzer
+ * @brief Module for analyzing stack usage in samples
+ *
+ * This module implements functions and the configuration that simplifies
+ * stack size analysis.
+ * @{
+ */
+
+/**
+ * @brief Stack size analyzer callback function
+ *
+ * Callback function with stack size statistics.
+ *
+ * @param name  The name of the thread
```

Comment:
or stringified address of the thread handle if name is not set.

Reply:
Comment added